src/publishers.rs

```diff
@@ -135,14 +170,12 @@ pub fn fetch_owners_of_crates(
                 i,
                 crates_io_names.len()
             );
-            users.insert(
-                crate_name.clone(),
-                publisher_users(&mut client, crate_name).unwrap(), //TODO: don't panic
-            );
-            teams.insert(
-                crate_name.clone(),
-                publisher_teams(&mut client, crate_name).unwrap(), //TODO: don't panic
-            );
+            if let Ok(pusers) = publisher_users(&mut client, crate_name) {
+                users.insert(crate_name.clone(), pusers);
+            }
+            if let Ok(pteams) = publisher_teams(&mut client, crate_name) {
+                teams.insert(crate_name.clone(), pteams);
+            }
```

Comment:
Doesn't this _omit_ users and teams that were unsuccessful, but appears to succeed? That sounds very unexpected or dangerous.

Reply:
The correct behavior would be retrying the request a few times (e.g. making 3 requests total) with exponential backoff on failure.

Reply:
I can rewrite to 3 tries with exponential backoff, that's no problem. 10-30-90 second waits? 

But now I'm not sure how we should handle the error case. You wrote that we shouldn't panic, so I took it as you wanted to keep going with the other ones. I need more information to fix this properly.

Reply:
The intent, I believe, was to fail but give a proper error message instead of panicking.

Reply:
Aah, that makes sense.

Is this alright?
```
Fetching data for "adler" (0/76)
Failed retrieving "https://crates.io/api/v1/crates/adler/owner_user", trying again in 10 seconds, attempt 1/3
Failed retrieving "https://crates.io/api/v1/crates/adler/owner_user", trying again in 30 seconds, attempt 2/3
Failed retrieving "https://crates.io/api/v1/crates/adler/owner_user", trying again in 90 seconds, attempt 3/3
Error Custom { kind: ConnectionReset, error: "Failed to retrieve publisher users" }
```
I looked into getting the error message but I'm not sure how? can't find a msg() or similar in https://doc.rust-lang.org/std/io/struct.Error.html

Reply:
For showing messages formatted for the user, you usually use the [`Display` impl](https://doc.rust-lang.org/std/io/struct.Error.html#impl-Display).

Reply:
For examples in simple cases: `format!("{}", io_error)` or `eprintln!("{}", io_error)`.

Reply:
_sigh_, I tried the Display impl, but I forgot to rebuild before running so I didn't actually try it :)

Reply:
`Failed retrieving "https://crates.io/api/v1/crates/adler/owner_user", trying again in 10 seconds, attempt 1/3`

This message is very nice! Makes it very clear what's happening.

Reply:
Thanks :)